### PR TITLE
Create new team

### PR DESCRIPTION
**Pull request by k3ntako**

### CODE REVIEW

**Comment by k3ntako** [comment]
I'm wondering if we shouldn't ask for a team email on create. I feel like that could slow people down, and we want people moving past that quickly?

File: server/controllers/TeamController.js [hunk 1]
```diff
@@ -71,6 +73,49 @@ const TeamController = {
     }
   },
 
+  async create(req, res, next){
+    try{
+      let { teamName, teamEmail } = req.body;
```
Comment (mienaikoe) [inline server/controllers/TeamController.js]: can we rename these to `name`, and `email`? We know they're for the team already by virtue of using this endpoint.

File: server/controllers/TeamController.js [hunk 2]
```diff
@@ -71,6 +73,49 @@ const TeamController = {
...
+
+      teamName = teamName.trim();
```
Comment (mienaikoe) [inline server/controllers/TeamController.js]: check for existence before calling trim. Something like `name = name && name.trim()`

File: server/controllers/TeamController.js [hunk 3]
```diff
@@ -71,6 +73,49 @@ const TeamController = {
...
+      if ( !teamName ){
+        throw new ForbiddenError("You must provide a team name");
+      }
+
+      // Validate team email
+      teamEmail = teamEmail.trim().toLowerCase(); // team email is stored as lowercase
```
Comment (mienaikoe) [inline server/controllers/TeamController.js]: same here. `email = email && email.trim().toLowerCase()`

File: server/controllers/TeamController.js [hunk 4]
```diff
@@ -71,6 +73,49 @@ const TeamController = {
...
+      res.json(newTeam);
+      next();
+    }catch( err ){
+      next(err);
+    }
+  },
+
   async update(req, res, next){
     try {
       const team = await Team.findOne({
```
Comment (mienaikoe) [inline server/controllers/TeamController.js]: This findOne call should be done after validations

File: server/controllers/TeamController.js [hunk 5]
```diff
@@ -115,12 +165,21 @@ const TeamController = {
...
   },
 
-  async checkEmailValid(req, res, next){
-    let { teamEmail } = req.body;
-    const isEmailValid = await TeamUtils.checkEmailValid(req.session.team_id, teamEmail);
+  async validateTeamEmail(req, res, next){
+    try{
+      const { teamEmail } = req.body;
+
+      if ( !req.session.team_id ){
```
Comment (mienaikoe) [inline server/controllers/TeamController.js]: `validateTeamEmail` might be called by someone who hasn't made a team yet. Remove this check.

File: client/models/Team.js [hunk 6]
```diff
@@ -10,13 +10,6 @@ class Team extends Model {
     super(props);
   }
 
-  static create( teamName, teamEmail ){
-    return Fetcher.post('/api/team/create', {
-      teamName,
-      teamEmail,
-    }).then( teamJson => new Team(teamJson));
-  }
-
```
Comment (mienaikoe) [inline client/models/Team.js]: Ah yea. I had realized we both wrote it, but it was late by the time I realized it and I was sleepy lol.

**Pull request merged**